src/colmap/estimators/similarity_transform.h

```diff
@@ -129,44 +121,48 @@ EstimateSim3dRobust(const std::vector<Eigen::Vector3d>& src,
 template <int kDim, bool kEstimateScale>
 void SimilarityTransformEstimator<kDim, kEstimateScale>::Estimate(
     const std::vector<X_t>& src,
-    const std::vector<Y_t>& dst,
+    const std::vector<Y_t>& tgt,
     std::vector<M_t>* models) {
-  THROW_CHECK_EQ(src.size(), dst.size());
+  THROW_CHECK_EQ(src.size(), tgt.size());
+  THROW_CHECK_GE(src.size(), kMinNumSamples);
   THROW_CHECK(models != nullptr);
 
   models->clear();
 
-  Eigen::Matrix<double, kDim, Eigen::Dynamic> src_mat(kDim, src.size());
-  Eigen::Matrix<double, kDim, Eigen::Dynamic> dst_mat(kDim, dst.size());
-  for (size_t i = 0; i < src.size(); ++i) {
-    src_mat.col(i) = src[i];
-    dst_mat.col(i) = dst[i];
+  using MatrixType = Eigen::Matrix<double, kDim, Eigen::Dynamic>;
+  const Eigen::Map<const MatrixType, Eigen::Aligned16> src_mat(
+      reinterpret_cast<const double*>(src.data()), 2, src.size());
```

Comment:
Maybe this is a dumb question. But I could not understand why here it is 2 rather than kDim?

Reply:
It's a bug :-) Thanks for spotting.